src/util/get-text.ts

```diff
@@ -41,18 +42,11 @@ export function currentFunction(editor: vscode.TextEditor): SelectionAndText {
 export function currentTopLevelFunction(editor: vscode.TextEditor): SelectionAndText {
     if (editor) {
         const document = editor.document;
-        const selection = select.getFormSelection(document, editor.selection.active, true);
-        const startPositionOfTopLevelForm = selection.start;
-        const cursorOffset = editor.document.offsetAt(startPositionOfTopLevelForm);
-        const tokenCursor = docMirror.getDocument(editor.document).getTokenCursor(cursorOffset);
-        if (tokenCursor.downList()) {
-            tokenCursor.forwardWhitespace();
-            while (tokenCursor.next()) {
-                const symbol = tokenCursor.getToken();
-                if (symbol.type === 'id') {
-                    return [selection, symbol.raw];
-                }
-            }
+        const cursorOffset = document.offsetAt(editor.selection.active);
+        const tokenCursor = docMirror.getDocument(document).getTokenCursor(cursorOffset);
+        const [range, text] = cursorTextGetter.currentTopLevelFunction(tokenCursor);
+        if (range) {
+            return [select.selectionFromOffsetRange(document, range), text]; // TODO: Figure out which selection to return...
```

Comment:
I don't understand this TODO. Is it still a todo? Looks like there's only one selection to return.

Reply:
Thanks. Yeah, we were previously returning the text of the function name, but the selection of the whole function. Which seemed really weird, but I didn't know if that was intentional. I don't think it was, and for this particular function no-one is using the selection so I fixed it. And forgot the TODO. 😄